ophyd/areadetector/base.py

```diff
@@ -4,6 +4,7 @@
 import sys
 from collections import OrderedDict
 import networkx as nx
+import matplotlib.pyplot as plt
```

Comment:
Ophyd [does not require matplotlib](https://github.com/NSLS-II/ophyd/blob/master/requirements.txt) so this import would introduce a new dependency. Since matplotlib is quite a heavy dependency and only needed for this one method, I think we should take a special approach: import it inside the method where it is used (i.e. Line 256). This is not usually good practice because it goes against the principle of failing early and clearly and it makes it harder for a developer to immediately see a module's dependencies. But for cases like this, it is a standard approach.

Would you add `matplotlib` to `test-requirements.txt` and add a quick test to [this file](https://github.com/NSLS-II/ophyd/blob/master/ophyd/tests/test_areadetector.py)? It can be a "smoke test" --- a test that just checks that a function runs without errors and doesn't bother to `assert` anything about its output.